docs/sphinx/sections/api/apidocs/dynamic.rst

```diff
@@ -8,3 +8,73 @@ These APIs provide the means for a simple kind of *dynamic orchestration* — wh
 .. autoclass:: DynamicOut
 
 .. autoclass:: DynamicOutput
+
+
+Mapping Details
+---------------
+
+To establish dependencies on a dynamic output in a composition function like :py:class:`job` or :py:class:`graph`, the ``map`` function is used on the object representing the dynamic output. `map` takes a ``Callable`` which receives a single argument. This callable is evaluated once, and the :py:class:`op` or :py:class:`graph` that are passed the input argument will establish dependencies. The return value is captured and wrapped in an object that allows for subsequent ``map`` or ``collect`` calls.
```

Comment:
I also find this statement generally hard to follow without an accompanying code snippet. I think the below snippets somewhat alleviate this, but I think a snippet showing a simple mapping in action would be helpful.